zboxcore/sdk/deleteworker.go

```diff
@@ -81,41 +86,55 @@ func (req *DeleteRequest) getObjectTreeFromBlobber(blobber *blockchain.StorageNo
 }
 
 func (req *DeleteRequest) ProcessDelete() error {
-	numList := len(req.blobbers)
-	objectTreeRefs := make([]fileref.RefEntity, numList)
+	num := len(req.blobbers)
+	objectTreeRefs := make([]fileref.RefEntity, num)
+	removedNum := 0
 	req.wg = &sync.WaitGroup{}
-	req.wg.Add(numList)
-	for i := 0; i < numList; i++ {
+	req.wg.Add(num)
+	for i := 0; i < num; i++ {
 		go func(blobberIdx int) {
 			defer req.wg.Done()
 			refEntity, err := req.getObjectTreeFromBlobber(req.blobbers[blobberIdx])
-			if err != nil {
-				Logger.Error(err.Error())
+			if err == nil {
+				req.Lock()
+				req.consensus.consensus++
+				req.listMask |= (1 << uint32(blobberIdx))
+				objectTreeRefs[blobberIdx] = refEntity
+				req.Unlock()
+				return
+			}
+			//it was removed from the blobber
+			if errors.Is(err, constants.ErrNotFound) {
+				req.Lock()
+				req.consensus.consensus++
+				removedNum++
+				req.Unlock()
+
 				return
 			}
-			req.consensus++
-			req.listMask |= (1 << uint32(blobberIdx))
-			objectTreeRefs[blobberIdx] = refEntity
+
+			Logger.Error(err.Error())
+
 		}(i)
 	}
 	req.wg.Wait()
 
 	req.deleteMask = uint32(0)
-	req.consensus = 0
+	req.consensus.consensus = float32(removedNum)
```

Comment:
If `It also is used in zboxutils.do callback.`, we'd better keep this in lock as well.

Reply:
`getObjectTree` requests are done. `delete` requests art not started yet. it should be unnecessary to do it with lock here. do you think so?